pkg/runtime/expressions/for_result.go

```diff
@@ -0,0 +1,89 @@
+package expressions
+
+import (
+	"github.com/MontFerret/ferret/pkg/runtime/core"
+	"github.com/MontFerret/ferret/pkg/runtime/values"
+)
+
+type ForResult struct {
+	itemList    *values.Array
+	hashTable   map[uint64]bool
+	distinct    bool
+	spread      bool
+	passThrough bool
+}
+
+func NewForResult(capacity int) *ForResult {
+	res := new(ForResult)
+	res.itemList = values.NewArray(capacity)
+
+	return res
+}
+
+func (f *ForResult) Distinct(distinct bool) *ForResult {
+	f.distinct = distinct
+
+	if f.distinct {
+		f.hashTable = make(map[uint64]bool)
+	} else {
+		f.hashTable = nil
+	}
+
+	return f
+}
+
+func (f *ForResult) Spread(spread bool) *ForResult {
+	f.spread = spread
+
+	return f
+}
+
+func (f *ForResult) PassThrough(passThrough bool) *ForResult {
+	f.passThrough = passThrough
+
+	return f
+}
+
+func (f *ForResult) Push(value core.Value) {
+	if f.passThrough {
+		return
+	}
+
+	if f.distinct {
+		// We need to check whether the value already exists in the result set
+		hash := value.Hash()
+		_, exists := f.hashTable[hash]
```

Comment:
`f.hashTable` has type `map[uint64]bool` and value always `true`, so this can be rewritten as 
```go
if f.hashTable[hash] {
    return
}
```

Reply:
If the key does not exist `f.hashTable[hash]` returns default value for bool which is `false`